src/com/amazon/ion/Timestamp.java

```diff
@@ -1341,6 +1360,37 @@ public static Timestamp forMillis(long millis, Integer localOffset)
         return new Timestamp(millis, localOffset);
     }
 
+    /**
+     * Returns a Timestamp that represents the point in time that is {@code seconds} from the unix epoch
+     * (1970-01-01T00:00:00.000Z), with the {@code nanoOffset} applied and time zone set to UTC.
+     *
+     * This function is intended to allow easy conversion to Timestamp from Java 8's {@code java.time.Instant}
+     * without having to port this library to Java 8.
+     *
+     * The following snippet will yield a Timestamp {@code ts} that equivalent to the {@code java.time.Instant}
+     * {@code i}:
+     *
+     * <code>
+     *     Instant i = Instant.now();
+     *     Timestamp ts = Timestamp.forEpochSecond(i.getEpochSecond(), i.getNano());
+     * </code>
+     *
+     * @param seconds The number of seconds from the unix epoch (1970-01-01T00:00:00.000Z).
+     * @param nanoOffset The number of nanoseconds for the fractional component. Must be between 0 and 999,999,999.
+     */
+    public static Timestamp forEpochSecond(long seconds, int nanoOffset) {
```

Comment:
This should probably accept a local offset, like `forMillis`.

Reply:
I was considering that.  Will add soon.